### PR TITLE
Allow the user to configure the range of DogStatsD metrics

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit introduces a configuration option to allow the user to define the range of values that appear in DogStatsD metrics. This range applies to all metric kinds. The distribution is changed from Standard -- "numerically uniform" -- to actually Uniform. We do not allow users to configure the distribution.

### Related issues

REF SMP-694

### CODE REVIEW

**Review by GeorgeHahn — APPROVED** [review]

File: lading_payload/src/dogstatsd.rs [hunk 1]
```diff
@@ -27,6 +27,14 @@ fn contexts_maximum() -> u16 {
     10_000
 }
 
+fn value_minimum() -> i64 {
+    i64::MIN
```
Comment (scottopell) [inline lading_payload/src/dogstatsd.rs:31]: I'd vote for more reasonable default values, this is known to be a worst-case for sketch calculation, right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+## [0.18.1-rc3]
+### Changed
+- DogStatsD NumValue is now int-only for experimental purposes, range is also made inclusive. The inclusivity will remain.
```
[the remark this answers is not in the document]
Comment (blt) [reply CHANGELOG.md:11]: Correct.

File: lading_payload/src/dogstatsd.rs (hunk 1 above)
Comment (blt) [reply lading_payload/src/dogstatsd.rs]: We now believe that the range is only part of it. If you send `1` rapidly enough you can force the distribution to allocate like mad. @GeorgeHahn will have a writeup when he's back from PTO.

File: lading_payload/src/dogstatsd.rs [hunk 3]
```diff
@@ -118,6 +128,30 @@ impl Default for MetricWeights {
     }
 }
 
+/// Configuration for the values of a metric.
+#[derive(Debug, Deserialize, Clone, PartialEq, Copy)]
+pub struct ValueConf {
+    /// Odds out of 256 that the value will be a float and not an integer.
+    float_weight: u8,
```
Comment (scottopell) [inline lading_payload/src/dogstatsd.rs:135]: This is a fairly unintuitive unit for "chance of thing occurring" (I came here from https://github.com/DataDog/single-machine-performance/pull/1268/files#diff-34dd14dde2466a88f431ddc27fc1d897ffede2f88d81eb37711a0d5886b0af92R29), especially given the prior art of `multipack_value_probability`.
What do you think of making this a 0-1 value and renaming to `probability`?

(Also open to renaming/changing `multipack_value_probability`, just looking for consistency between these two options really)

Comment (blt) [reply lading_payload/src/dogstatsd.rs:135]: Ah, that's a good call-out. I'd rather this be consistent.

**Pull request merged**